searchlib/src/vespa/searchlib/common/bitvector.cpp

```diff
@@ -177,9 +177,18 @@ BitVector::countInterval(Index start, Index end) const
 void
 BitVector::orWith(const BitVector & right)
 {
-    verifyContains(*this, right);
-    IAccelrated::getAccelrator().orBit(getActiveStart(), right.getWordIndex(getStartIndex()), getActiveBytes());
+    verifyInclusiveStart(*this, right);
 
+    if (right.size() < size()) {
+        ssize_t commonBytes = numActiveBytes(getStartIndex(), right.size()) - sizeof(Word);
+        if (commonBytes > 0) {
+            IAccelrated::getAccelrator().orBit(getActiveStart(), right.getWordIndex(getStartIndex()), commonBytes);
+        }
+        Index last(right.size() - 1);
```

Comment:
If right.size() is 0 then we get a wraparound and bad memory accesses. Note that bitvector have other methods that fails in a similar manner when size() is 0, e.g. clearIntervalNoInvalidation(), setInterval(), countInterval().

Reply:
Guard added